gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/DocumentReferenceTest.java

```diff
@@ -0,0 +1,173 @@
+        assertThat(documentReference.getIndexedElement().getValue().toInstant().toString()).isEqualTo("2010-01-14T00:00:00Z");
+        assertThat(documentReference.getCreatedElement().getValue().toInstant().toString()).isEqualTo("2020-10-12T13:33:44Z");
+    }
+
+    private void assertOptionalValidData(DocumentReference documentReference) {
+        assertThat(documentReference.getId()).isEqualTo(NARRATIVE_STATEMENT_ROOT_ID);
+        assertThat(documentReference.getMeta().getProfile().get(0).getValue()).isEqualTo(META_PROFILE);
+        assertThat(documentReference.getStatus()).isEqualTo(Enumerations.DocumentReferenceStatus.CURRENT);
+        assertThatIdentifierIsValid(documentReference.getIdentifierFirstRep(), documentReference.getId());
+        assertThat(documentReference.getType().getText()).isEqualTo(NARRATIVE_STATEMENT_TYPE);
+        assertThat(documentReference.getAuthor().get(0).getReference()).isEqualTo("Practitioner/2D70F602-6BB1-47E0-B2EC-39912A59787D");
+        assertThat(documentReference.getDescription()).isNull();
+        assertThat(documentReference.getCreated()).isNull();
+        assertThat(documentReference.getContent()).isEmpty();
+    }
+
+    private void assertAttachmentData(DocumentReference documentReference) {
+        assertThat(documentReference.getContent().get(0).getAttachment().getTitle()).isEqualTo(FILENAME);
+        assertThat(documentReference.getContent().get(0).getAttachment().getUrl()).isEqualTo(URL);
+        assertThat(documentReference.getContent().get(0).getAttachment().getContentType()).isEqualTo(CONTENT_TYPE);
+    }
+
+    private void assertDocumentReferenceWithAbsentAttachment(DocumentReference documentReference) {
+        assertThat(documentReference.getId()).isEqualTo(NARRATIVE_STATEMENT_ROOT_ID);
+        assertThat(documentReference.getContent().get(0).getAttachment().getTitle()).isEqualTo(PLACEHOLDER);
+        assertThat(documentReference.getContent().get(0).getAttachment().getUrl()).isNull();
+        assertThat(documentReference.getContent().get(0).getAttachment().getContentType()).isEqualTo(CONTENT_TYPE);
+    }
+
+    private void assertThatIdentifierIsValid(Identifier identifier, String id) {
+        assertThat(identifier.getSystem()).isEqualTo(IDENTIFIER_SYSTEM);
+        assertThat(identifier.getValue()).isEqualTo(id);
+    }
+
+    private static Patient createPatient() {
+        Patient patient = new Patient();
+        patient.setId(randomUUID().toString());
+        return patient;
+    }
+
+    private void setUpCodeableConceptMock() {
+        var codeableConcept = new CodeableConcept();
+        var coding = new Coding();
+        coding.setDisplay(CODING_DISPLAY);
+        codeableConcept.addCoding(coding);
+        when(codeableConceptMapper.mapToCodeableConcept(any())).thenReturn(codeableConcept);
+    }
+
+    private List<Encounter> getEncounterList() {
```

Comment:
you should be asserting the context field contains this encounter id. Also should add a test for if there is no matching encounter id - i.e the ehrComposition id is not equal to the ENCOUNTER_ID